cmake/OpenCVUtils.cmake

```diff
@@ -1423,6 +1423,15 @@ macro(ocv_copyfiles_add_target target list_var comment_str)
   add_custom_target(${target} DEPENDS "${OPENCV_DEPHELPER}/${target}")
 endmacro()
 
+macro(ocv_get_smart_file_name output_var fpath)
+  if("${fpath}" MATCHES "^(${OpenCV_BINARY_DIR})([/\\]?)(.*)$")
+    set(${output_var} "<BUILD>/${CMAKE_MATCH_3}")
```

Comment:
`file(RELATIVE_PATH <variable> <directory> <file>)` can be used here instead of captured value.